arch/csr/hstateen0.yaml

```diff
@@ -0,0 +1,100 @@
+  the same bit appears as read-only zero in sstateen when accessed in VS-mode.
+
+  A bit in an hstateen CSR cannot be read-only one unless the same bit is read-only one
+  in the matching mstateen CSR.
+
+definedBy:
+  allOf:
+    - H
+    - Smstateen
+    - Ssstateen
+fields:
+  SEO:
+    location: 63
+    description: |
+      The SE0 bit in hstateen0 controls access to the sstateen0 CSR.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  ENVCFG:
+    location: 62
+    description: |
+      The ENVCFG bit in hstateen0 controls access to the senvcfg CSRs.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  CSRIND:
+    location: 60
+    description: |
+      The CSRIND bit in hstateen0 controls access to the siselect and the
+      sireg*, (really vsiselect and vsireg*) CSRs provided by the Sscsrind
+      extensions.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  AIA:
+    location: 59
+    description: |
+      The AIA bit in hstateen0 controls access to all state introduced by
+      the Ssaia extension and is not controlled by either the CSRIND or the
+      IMSIC bits of hstateen0.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  IMSIC:
+    location: 58
+    description: |
+      The IMSIC bit in hstateen0 controls access to the guest IMSIC state,
+      including CSRs stopei (really vstopei), provided by the Ssaia extension.
+
+      Setting the IMSIC bit in hstateen0 to zero prevents a virtual machine
+      from accessing the hart’s IMSIC the same as setting `hstatus.`VGEIN = 0.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  CONTEXT:
+    location: 57
+    description: |
+      The CONTEXT bit in hstateen0 controls access to the scontext CSR provided
+      by the Sdtrig extension.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  JVT:
+    location: 2
+    description: |
+      The JVT bit controls access to the jvt CSR provided by the Zcmt extension.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  FCSR:
+    location: 1
+    description: |
+      The FCSR bit controls access to fcsr for the case when floating-point instructions
+      operate on x registers instead of f registers as specified by the Zfinx and related
+      extensions (Zdinx, etc.). Whenever misa.F = 1, FCSR bit of mstateen0 is read-only
+      zero (and hence read-only zero in hstateen0 and sstateen0 too). For convenience,
+      when the stateen CSRs are implemented and misa.F = 0, then if the FCSR bit of a
+      controlling stateen0 CSR is zero, all floating-point instructions cause an illegal
+      instruction trap (or virtual instruction trap, if relevant), as though they all access
+      fcsr, regardless of whether they really do.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+  C:
+    location: 0
+    description: |
+      The C bit controls access to any and all custom state. This bit is not custom state itself.
+      The C bit of these registers is not custom state itself; it is a standard field of a
```

Comment:
Oddly redundant, but that's what the spec has. I guess we leave it for now. Interested in opening an issue against the spec?

Reply:
Here's what the spec says today:
> The C bit controls access to any and all custom state.
> The C bit of these registers is not custom state itself; it is a
> standard field of a standard CSR, either `mstateen0`, `hstateen0`, or
> `sstateen0`.

Could you use that text here, instead?